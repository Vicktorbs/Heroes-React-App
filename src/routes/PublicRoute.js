import React from 'react'
import PropTypes from 'prop-types';
import { Redirect, Route } from 'react-router-dom'

export const PublicRoute = ({
    isNotAuthenticated,
    component: Component,
    ...rest
}) => {
    return (
        <Route { ...rest }
            component={ (props) => (
                (isNotAuthenticated) ?
                    (<Component { ...props } />) :
                    (<Redirect to='/' />)
            )}
        />
    )
}

PublicRoute.protoTypes = {
    isNotAuthenticated: PropTypes.bool.isRequired,
    component: PropTypes.func.isRequired
}
