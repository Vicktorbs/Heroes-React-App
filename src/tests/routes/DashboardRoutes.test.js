import React from 'react';
import { mount } from "enzyme"
import { DashboardRoutes } from "../../routes/DashboardRoutes"
import { AuthContext } from '../../auth/AuthContext';
import { MemoryRouter } from 'react-router-dom';

describe('Test on Router <DashboardRoutes />', () => {
    
    test('should show itself correctly', () => {
        const contexValue = {
            dispatch: jest.fn(),
            user: {
                logged: true,
                name: 'Víctor'
            }
        }
        const wrapper = mount(
            <AuthContext.Provider value={ contexValue } >
                <MemoryRouter>
                    <DashboardRoutes />

                </MemoryRouter>
            </AuthContext.Provider>
        );
        expect(wrapper).toMatchSnapshot();
        expect(wrapper.find('.text-info').text().trim()).toBe('Víctor');
    })
    
})
