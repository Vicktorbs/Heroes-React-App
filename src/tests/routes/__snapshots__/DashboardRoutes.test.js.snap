// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test on Router <DashboardRoutes /> should show itself correctly 1`] = `
Array [
  <nav
    className="navbar navbar-expand-sm navbar-dark bg-dark"
  >
    <a
      className="navbar-brand"
      href="/"
      onClick={[Function]}
    >
      Asociaciones
    </a>
    <div
      className="collapse navbar-collapse"
    >
      <div
        className="navbar-nav me-auto mb-2 mb-lg-0"
      >
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/marvel"
          onClick={[Function]}
        >
          Marvel
        </a>
        <a
          aria-current="page"
          className="nav-item nav-link active"
          href="/dc"
          onClick={[Function]}
          style={Object {}}
        >
          DC
        </a>
        <a
          aria-current={null}
          className="nav-item nav-link"
          href="/search"
          onClick={[Function]}
        >
          Search
        </a>
      </div>
      <span
        className="navbar-text text-info"
      >
        VÃ­ctor
      </span>
      <span
        className="navbar-text"
      >
        <button
          className="nav-item nav-link btn text-white"
          onClick={[Function]}
        >
          Logout
        </button>
      </span>
    </div>
  </nav>,
  <div
    className="container mt-3"
  >
    <div>
      <h3>
        DC!
      </h3>
      <div
        className="row row-cols-1 row-cols-md-3 g-4 mb-3 animate__animated animate__bounce animate__fadeIn"
      >
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Batman"
                  className="card-img-top"
                  src="./assets/heroes/dc-batman.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Batman
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Bruce Wayne
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Detective Comics #27
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-batman"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Superman"
                  className="card-img-top"
                  src="./assets/heroes/dc-superman.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Superman
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Kal-El
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Action Comics #1
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-superman"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Flash"
                  className="card-img-top"
                  src="./assets/heroes/dc-flash.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Flash
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Jay Garrick
                  </p>
                  <p
                    className="card-text"
                  >
                    Jay Garrick, Barry Allen, Wally West, Bart Allen
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Flash Comics #1
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-flash"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Green Lantern"
                  className="card-img-top"
                  src="./assets/heroes/dc-green.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Green Lantern
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Alan Scott
                  </p>
                  <p
                    className="card-text"
                  >
                    Alan Scott, Hal Jordan, Guy Gardner, John Stewart, Kyle Raynor, Jade, Sinestro, Simon Baz
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      All-American Comics #16
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-green"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Green Arrow"
                  className="card-img-top"
                  src="./assets/heroes/dc-arrow.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Green Arrow
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Oliver Queen
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      More Fun Comics #73
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-arrow"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Wonder Woman"
                  className="card-img-top"
                  src="./assets/heroes/dc-wonder.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Wonder Woman
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Princess Diana
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      All Star Comics #8
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-wonder"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Martian Manhunter"
                  className="card-img-top"
                  src="./assets/heroes/dc-martian.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Martian Manhunter
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    J'onn J'onzz
                  </p>
                  <p
                    className="card-text"
                  >
                    Martian Manhunter
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Detective Comics #225
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-martian"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Robin/Nightwing"
                  className="card-img-top"
                  src="./assets/heroes/dc-robin.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Robin/Nightwing
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Dick Grayson
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Detective Comics #38
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-robin"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Blue Beetle"
                  className="card-img-top"
                  src="./assets/heroes/dc-blue.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Blue Beetle
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Dan Garret
                  </p>
                  <p
                    className="card-text"
                  >
                    Dan Garret, Ted Kord, Jaime Reyes
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Mystery Men Comics #1
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-blue"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
        <div
          className="col"
        >
          <div
            className="card"
          >
            <div
              className="row"
            >
              <div
                className="col-md-4"
              >
                <img
                  alt="Black Canary"
                  className="card-img-top"
                  src="./assets/heroes/dc-black.jpg"
                />
              </div>
              <div
                className="col-md-8"
              >
                <div
                  className="card-body"
                >
                  <h5
                    className="card-title"
                  >
                    Black Canary
                  </h5>
                  <p
                    className="card-text text-muted"
                  >
                    Dinah Drake
                  </p>
                  <p
                    className="card-text"
                  >
                    Dinah Drake, Dinah Lance
                  </p>
                  <p
                    className="card-text"
                  >
                    <small
                      className="text-muted"
                    >
                      Flash Comics #86
                    </small>
                  </p>
                  <a
                    className="btn btn-primary"
                    href="/hero/dc-black"
                    onClick={[Function]}
                  >
                    More...
                  </a>
                </div>
              </div>
            </div>
          </div>
        </div>
      </div>
    </div>
  </div>,
]
`;
